### PR TITLE
user-order : create user order

### PR DESCRIPTION
**Pull request by kechyy**
#### Description
 - This PR implements cleanit user order request

#### Type of Change
- [x] Feature(Create user order request)

#### How Has This Been Tested?
- [ ] Clone the repo
- [ ] On your terminal, run npm install
- [ ] On your terminal, run npm serve

#### Checklist:
- [x] Create user order request firebase function controller
- [x] Create middleware for order request input validation

#### Relevant Trello Tickets
[#qdtKlxB4](https://trello.com/c/qdtKlxB4)

### CODE REVIEW

**Comment by Segun-Ade** [comment]
LGTM

File: functions/controllers/order.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                    createdAt
+                });
+            }
+            return res.json({
+                message: 'Order was successfully created',
+                data: orderID,
+            }); 
+        } 
+    } catch(error) {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
```
Comment (enyeInc) [inline functions/controllers/order.js]: destrcuture `INTERNAL_SERVER_ERROR` from `HttpStatus`

File: functions/middleware/order.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const HttpStatus = require('http-status-codes');
+const { BAD_REQUEST } = HttpStatus;
+exports.orderValidation =  (req, res, next) => {
+    const { vendorID, vendorLocation, orderItems, orderDetails, totalPrice } = req.body;
+    if(!vendorID && !vendorLocation && !orderItems && !orderDetails && totalPrice ) {
```
Comment (enyeInc) [inline functions/middleware/order.js]: this check is very confusing. Define it as a variable first before using it, and make the variable name verbose

File: functions/utils/index.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+exports.randomID = () => {
```
Comment (enyeInc) [inline functions/utils/index.js]: use [JSDocs](https://jsdoc.app/about-getting-started.html) when commenting your functions

**Comment by Way2nnadi** [comment]
@kechyy Make the changes and merge this code down

**Pull request merged**